pinc/ProjectTransition.inc

```diff
@@ -462,120 +453,474 @@ function get_transition($from_state, $to_state)
 // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
-new ProjectTransition(
-    PROJ_NEW,
-    PROJ_P1_UNAVAILABLE,
-    'unrestricted_manager',
-    [
-        'project_restriction' => null,
-        'action_name' => '[default]',
-        'confirmation_question' => null,
-        'detour' => null,
-        'settings_template' => '',
-        'collateral_actions' => null,
-    ]
-);
+function create_all_transitions()
```

Comment:
I like the idea that we don't create them unless we actually need them. And also that we're storing the set when we create them so we don't create them more than once.

Given the PR with the new Activities, Rounds, etc classes, should we (very lightly) mirror that here? Something like:
```php
class ProjectTransitions
    private static $_project_transitions = [];

    public static get_all()
    {
        if (!self::$_project_transitions) {
            self::$_project_transitions = create_project_transitions();
        }
        return self::$_project_transitions;
    }
}
```
And have this function just return the array of transitions and the use of `$PROJECT_TRANSITIONS` becomes `ProjectTransitions::get_all()`.

Reply:
That makes it seem more sensible, rather than having a function called `create` that only creates if they don't already exist. I'll make it so.